src/engine/pieces/pawn.ts

```diff
@@ -1,13 +1,47 @@
 import Piece from './piece';
 import Player from '../player';
 import Board from '../board';
+import Square from '../square';
+import King from "./king";
 
 export default class Pawn extends Piece {
     public constructor(player: Player) {
         super(player);
     }
 
     public getAvailableMoves(board: Board) {
-        return new Array(0);
+        let moves: Square[] = [];
+        let currentSquare = board.findPiece(this);
+        let direction = (this.player === Player.WHITE) ? 1 : -1;
+
+        let squareAhead = Square.at(currentSquare.row + direction, currentSquare.col);
+        if (squareAhead.row >= 0 && squareAhead.row < 8 && !board.getPiece(squareAhead)) {
```

Comment:
Poti extrage verificarea pentru in board bounds intr-o functie separata, e folosita la toate piesele.